iceoryx_utils/include/iceoryx_utils/internal/units/duration.hpp

```diff
@@ -37,127 +37,206 @@ class Duration;
+/// @brief Constructs a new Duration object from hours
 constexpr Duration operator"" _h(unsigned long long int); // PRQA S 48
 
-/// @brief constructs a new Duration object in days
-constexpr Duration operator"" _d(long double);
-
-/// @brief constructs a new Duration object in days
+/// @brief Constructs a new Duration object from days
 constexpr Duration operator"" _d(unsigned long long int); // PRQA S 48
 } // namespace duration_literals
 
 /// @code
 ///   #include <iostream>
 ///   // ...
 ///   using namespace units;
-///   auto timeInDays = 12_d;
-///   auto timeInSeconds = 1.5_s;
-///   std::cout << timeInDays << std::endl;
-///   std::cout << timeInDays.nanoSeconds<int>() << " ns" << std::endl;
-///   std::cout << timeInSeconds.minutes<float>() << " min" << std::endl;
+///   using namespace units::duration_literals;
+///   auto someDays = 2 * 7_d + 5_ns;
+///   auto someSeconds = 42_s + 500_ms;
+///   std::cout << someDays << std::endl;
+///   std::cout << someDays.nanoSeconds<uint64_t>() << " ns" << std::endl;
+///   std::cout << someSeconds.milliSeconds<int64_t>() << " ms" << std::endl;
 /// @endcode
 class Duration
 {
   public:
+    // BEGIN CREATION FROM STATIC FUNCTIONS
+
+    /// @brief Constructs a new Duration object from nanoseconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as nanoseconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration nanoseconds(const T ns);
+    static constexpr Duration nanoseconds(const T value);
+
+    /// @brief Constructs a new Duration object from microseconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as microseconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration microseconds(const T us);
+    static constexpr Duration microseconds(const T value);
+
+    /// @brief Constructs a new Duration object from milliseconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as milliseconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration milliseconds(const T ms);
+    static constexpr Duration milliseconds(const T value);
+
+    /// @brief Constructs a new Duration object from seconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as seconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration seconds(const T seconds);
+    static constexpr Duration seconds(const T value);
```

Comment:
We have also a `seconds` getter method. I would like to rename the creation methods in a follow up PR and add a `from` prefix, e.g. `fromSeconds` and make the getter methods all lower case.

It would be different from `chrono`, but the alternative would be to have something like a `units_duration_cast`, which I wouldn't like that much all over the code base.

Any objections?